### PR TITLE
Move alias type inference (experimental) behind --infer/-i flag

### PR DESCRIPTION
**Pull request by bailey-layzer**
"Fixes" #2416: aliases should work as previously unless -i flag is set.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -367,18 +367,17 @@ dependencies = [
 
 [[package]]
 name = "battery"
-version = "0.7.5"
+version = "0.7.6"
```
Comment (thegedge) [inline Cargo.lock:370]: We should limit dependency updates to PRs that focus solely on updates. Would you be able to revert the changes here?

File: crates/nu-cli/src/commands/alias.rs [hunk 2]
```diff
@@ -93,10 +96,14 @@ pub async fn alias(
...
-            .replace("-s", "");
+            .replace("-si", "-i")
+            .replace("-s ", "")
+            .replace("-is", "-i");
         let right = raw_input[right_brace..]
             .replace("--save", "")
-            .replace("-s", "");
+            .replace("-si", "-i")
+            .replace("-s ", "")
+            .replace("-is", "-i");
```
Comment (thegedge) [inline crates/nu-cli/src/commands/alias.rs:106]: Oooof, this will get tough to manage if we ever wanted to add more flags (although, I understand `--infer` is temporary). Just want to point out that we should consider a save mechanism that doesn't need to do this.

/cc @jonathandturner since you've done a bunch of work on aliases

File: Cargo.lock (hunk 1 above)
Comment (bailey-layzer) [reply Cargo.lock:370]: Whoops, missed that I'd changed that.

File: crates/nu-cli/src/commands/alias.rs (hunk 2 above)
Comment (jonathandturner) [reply crates/nu-cli/src/commands/alias.rs:106]: Thanks for the heads up. Yeah, I'll try to keep this in mind as we improve alias further.

**Comment by jonathandturner** [comment]
Thanks!
